crates/bazed-core/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+#![forbid(unreachable_pub)]
+use color_eyre::Result;
+use marked_rope::{MarkedRope, MarkerId};
+use std::{fs::File, io::BufReader, path::PathBuf};
+
+use ropey::Rope;
+
+pub mod marked_rope;
+
+pub struct Document {
+    source: Option<PathBuf>,
+    dirty: bool,
+    content: MarkedRope,
+}
```

Comment:
Does it make sense to include a separate data structure for portions already? Documents sure are cool and the final stuff (ontologically) we edit.

Reply:
I don't think so, for now we should focus on getting something that works, and then add Portions and all that sort of stuff once we have the relevant core architecture _roughly_ figured out